src/components/atom_preference_language_checkbox/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { putPreferenceApi } from '@/api/preferences/put-preference.api'
+import { getLanguageFullName } from '@/global.constants'
+import { GlobalLanguageCode } from '@/global.interface'
+import { preferenceState } from '@/recoil/preferences/preference.state'
+import { Checkbox, FormControlLabel } from '@mui/material'
+import { FC } from 'react'
+import { useRecoilCallback, useRecoilValue } from 'recoil'
+
+interface Props {
+  languageCode: GlobalLanguageCode
+}
+
+const PreferenceLanguageCheckbox: FC<Props> = ({ languageCode }) => {
+  const preference = useRecoilValue(preferenceState)
+  const onChange = useRecoilCallback(
+    ({ set, snapshot }) =>
+      async (_, checked: boolean) => {
+        try {
+          const preference = await snapshot.getPromise(preferenceState)
+          if (!preference) return
+
+          const nativeLanguagesSet = new Set([
+            ...preference.nativeLanguages,
+            languageCode,
+          ])
+
+          if (!checked) nativeLanguagesSet.delete(languageCode)
+          const nativeLanguages: GlobalLanguageCode[] =
+            Array.from(nativeLanguagesSet)
+
+          const [data] = await putPreferenceApi({ nativeLanguages })
+          set(preferenceState, data)
+        } catch {}
+      },
+    [languageCode],
+  )
```

Comment:
Make two hooks `usePutPreference` and `usePutPreferenceNativeLanguages`
- usePutPreference
  - simply modify the given preference and apply into the recoil
- usePutPreferenceNativeLanguages
  - get the native language and if it is checked and use the `usePutPreference` to actually apply